app/bundles/CoreBundle/Assets/js/1.core.js

```diff
@@ -72,6 +72,16 @@ mQuery( document ).ready(function() {
             e.preventDefault();
         }
     });
+
+    // Try to keep alive the session only if csrf-mautic_ajax_post cookie exists
+    if (mQuery.cookie('csrf-mautic_ajax_post')) {
```

Comment:
I found out that this snippet was being called when the user was not logged in too. I think adding a check for this by checking the CSRF cookies.

Reply:
That cookie doesn't exist, only here does it exist: https://github.com/mautic/mautic/pull/13745

I think any secure page (with /s/) gets a session and csrf token regardless of login status. But I'm not sure any such pages are accessible to anonymous users besides /s/login. 

Basically, afaik, if that script gets loaded there will be a csrf token available. Be it a variable as in 5.x or a cookie as in the PR.

You could check for specifically /s/login as current path? Or leave it as is. Unless you found other paths where it is running where it shouldn't?

Reply:
I agree that looking for the CSRF token cookie was not the right approach. I found that it was indeed from the server.https://github.com/mautic/mautic/pull/13745.

Upon investigating further, I found that yes, the `/s/login` page is the only one with anonymous user access.
And adding proper checks around that helps to eliminate any updated call to /keep-alive route.